NickvisionTagger.Shared/Models/MusicFile.cs

```diff
@@ -527,127 +525,189 @@ public bool RemoveCustomProperty(string name)
+    }
+    
+    /// <summary>
+    /// Uses the music file's tag to set the file's filename in the format of the format string
+    /// </summary>
+    /// <param name="formatString">Th format string</param>
+    /// <returns>True if successful, else false</returns>
+    public bool TagToFilename(string formatString)
+    {
+        if(string.IsNullOrEmpty(formatString))
+        {
+            return false;
+        }
+        var validProperties = new string[] { "title", _("title"), "artist", _("artist"), "album", _("album"), "year", _("year"), "track", _("track"), "albumartist", _("albumartist"), "genre", _("genre"), "comment", _("comment"), "composer", _("composer"), "description", _("description"), "publisher", _("publisher") };
+        var customProps = _customProperties.Keys.ToList();
+        var matches = Regex.Matches(formatString, @"%(\w+)%", RegexOptions.IgnoreCase); //wrapped in %%
+        if(matches.Count == 0)
         {
             return false;
         }
+        foreach(Match match in matches)
+        {
+            string value = match.Value.Remove(0, 1); //remove first %
+            value = value.Remove(value.Length - 1, 1); //remove last %;
+            if(validProperties.Contains(value.ToLower()))
+            {
+                value = value.ToLower();
+                var replace = "";
+                if(value == "title" || value == _("title"))
+                {
+                    replace = Title;
+                }
+                else if(value == "artist" || value == _("artist"))
+                {
+                    replace = Artist;
+                }
+                else if(value == "album" || value == _("album"))
+                {
+                    replace = Album;
+                }
+                else if(value == "year" || value == _("year"))
+                {
+                    replace = Year.ToString();
+                }
+                else if(value == "track" || value == _("track"))
+                {
+                    replace = Track.ToString("D2");
+                }
+                else if(value == "albumartist" || value == _("albumartist"))
+                {
+                    replace = AlbumArtist;
+                }
+                else if(value == "genre" || value == _("genre"))
+                {
+                    replace = Genre;
+                }
+                else if(value == "comment" || value == _("comment"))
+                {
+                    replace = Comment;
+                }
+                else if(value == "composer" || value == _("composer"))
+                {
+                    replace = Composer;
+                }
+                else if(value == "description" || value == _("description"))
+                {
+                    replace = Description;
+                }
+                else if(value == "publisher" || value == _("publisher"))
+                {
+                    replace = Publisher;
+                }
```

Comment:
you could use switch here (I know you love switches 😄)

Reply:
switch, if, they are the same too me hahaha...but `switch expression` that's something different and I like those hahah let me refactor...

Reply:
oh now I remember why I didn't use `switch expression`...you can't have multiple conditions for the same result:
```csharp
var replace = value switch
{
    "title" or _("title") => Title,
    ...
};
```
doesn't work, so it'd have to be something like:
```csharp
var replace = value switch
{
    "title" => Title,
    _("title") => Title,
    ...
};
```
which doesn't even work because it doesn't understand the `_()` for some reason so....`if` it is.

Reply:
ah okay, so you can't set condition to be a function call, only literal, got it

Reply:
yes `switch` and `switch expressions` require concrete values to work...string literals (i.e. `"sdsf"`) are concrete as they are immutable (a.k.a one instance of them exist and is reused throughout)